htdocs/product/list.php

```diff
@@ -1317,43 +1318,43 @@
 			}
 		}
 
-		$product_static->id = $obj->rowid;
-		$product_static->ref = $obj->ref;
-		$product_static->ref_fourn = empty($obj->ref_supplier) ? '' : $obj->ref_supplier; // deprecated
-		$product_static->ref_supplier = empty($obj->ref_supplier) ? '' : $obj->ref_supplier;
-		$product_static->label = $obj->label;
-		$product_static->finished = $obj->finished;
-		$product_static->type = $obj->fk_product_type;
-		$product_static->status_buy = $obj->tobuy;
-		$product_static->status     = $obj->tosell;
-		$product_static->status_batch = $obj->tobatch;
-		$product_static->entity = $obj->entity;
-		$product_static->pmp = $obj->pmp;
-		$product_static->accountancy_code_sell = $obj->accountancy_code_sell;
-		$product_static->accountancy_code_sell_export = $obj->accountancy_code_sell_export;
-		$product_static->accountancy_code_sell_intra = $obj->accountancy_code_sell_intra;
-		$product_static->accountancy_code_buy = $obj->accountancy_code_buy;
-		$product_static->accountancy_code_buy_intra = $obj->accountancy_code_buy_intra;
-		$product_static->accountancy_code_buy_export = $obj->accountancy_code_buy_export;
-		$product_static->length = $obj->length;
-		$product_static->length_units = $obj->length_units;
-		$product_static->width = $obj->width;
-		$product_static->width_units = $obj->width_units;
-		$product_static->height = $obj->height;
-		$product_static->height_units = $obj->height_units;
-		$product_static->weight = $obj->weight;
-		$product_static->weight_units = $obj->weight_units;
-		$product_static->volume = $obj->volume;
-		$product_static->volume_units = $obj->volume_units;
-		$product_static->surface = $obj->surface;
-		$product_static->surface_units = $obj->surface_units;
-		if (!empty($conf->global->PRODUCT_USE_UNITS)) {
-			$product_static->fk_unit = $obj->fk_unit;
-		}
+		$product_static->fetch($obj->rowid);
```

Comment:
Using a fetch guarantee that the fetch is complete, however, if we have a list of 500 lines, it means we will make 501 request to show the page instead of 1.
Such change makes code cleaner and easier but make application no more usable on large databases.

Reply:
@eldy what do you think of adding hook there ?